### ISSUE TITLE
User can manipulate failing bot to be active

### ISSUE THREAD

**Issue by remarkablerocket** [inglesp/botany#44]
Reproduce using django shell:

```
$ python manage.py shell
>>> from botany.actions import create_user, create_bot, set_bot_active, mark_bot_failed
>>> test_user = create_user("test-user@example.com", "Test User")
>>> bot1 = create_bot(test_user, "todo.py", "# TODO")
>>> mark_bot_failed(bot1)
>>> bot1.state
'failed'
>>> set_bot_active(bot1, test_user)
Traceback [...] assert bot.is_under_probation or bot.is_inactive [...]
>>> bot1.state
'failed'
>>> bot2 = create_bot(test_user, "todo.py", "# TODO")
>>> set_bot_active(bot2, test_user)
>>> bot1.refresh_from_db()
>>> bot1.state
'inactive'
>>> set_bot_active(bot1, test_user)
>>> bot1.state
'active'
```

I believe this process can be replicated via the website's UI.

Should be simple to fix, I'll have a look at it later

**Comment by remarkablerocket** [comment]
Please ignore my personal git journey above. PRs, branching and rebasing are fairly new to me :)

**Comment by remarkablerocket** [comment]
Fixed now #45 has been merged - closing
